assets/js/modules/adsense/common/use-snippet-switch.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * AdSense Use Snippet Switch component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import Switch from '../../../components/switch';
+import SettingsNotice from '../../../components/settings-notice';
+import { trackEvent } from '../../../util';
+import { STORE_NAME } from '../datastore';
+const { useSelect, useDispatch } = Data;
+
+export default function UseSnippetSwitch( props ) {
+	const {
+		label = __( 'Let Site Kit place code on your site', 'google-site-kit' ),
+		checkedMessage,
+		uncheckedMessage,
+		saveOnChange,
+	} = props;
+
+	const useSnippet = useSelect( ( select ) => select( STORE_NAME ).getUseSnippet() );
+	const isDoingSaveUseSnippet = useSelect( ( select ) => select( STORE_NAME ).isDoingSaveUseSnippet() );
+
+	const { setUseSnippet, saveUseSnippet } = useDispatch( STORE_NAME );
+	const onChange = useCallback( async () => {
+		setUseSnippet( ! useSnippet );
+		trackEvent( 'adsense_setup', useSnippet ? 'adsense_tag_enabled' : 'adsense_tag_disabled' );
+		if ( saveOnChange ) {
+			await saveUseSnippet();
+		}
+	}, [ useSnippet ] );
+
+	if ( 'undefined' === typeof useSnippet ) {
+		return null;
+	}
+
+	return (
+		<Fragment>
+			<div className="googlesitekit-setup-module__switch">
+				<Switch
+					label={ label }
+					onClick={ onChange }
+					checked={ useSnippet }
+					disabled={ isDoingSaveUseSnippet }
+					hideLabel={ false }
+				/> <span className="googlesitekit-recommended">{ __( 'RECOMMENDED', 'google-site-kit' ) }</span>
```

Comment:
Why is this text uppercase like this? If 100% desired we should do that with CSS, because this will confuse some screen readers (they might read this as an acronym rather than a word). Plus: it's not very friendly to dyslexic users and others who find all-caps text harder-to-scan.

Still: at the very least it's not semantically all-caps, so we should do this with CSS.

Reply:
I had a similar thought here but I recently learned that some screen readers may still see it as all caps if transformed with CSS. I agree that doing with CSS is probably the better way to go here though 👍

Reply:
Yeah, I think doing `small-caps` or something else is better still, but ultimately any changes in case has the potential to mess with assistive technology. All-caps is generally not very accessible 😬